start-vswitch.sh

```diff
@@ -4,54 +4,243 @@
 # -use multi-queue (for DPDK)
 # -bind DPDK PMDs to optimal CPUs
 # -for pvp topologies, provide a list of optimal cpus available for the VM
-# configure a network overlay, like VxLAN
+# configure an overlay network such as VxLAN
 
 # The following features are not implemented but would be nice to add:
 # -for ovs, configure x flows
 # -for a router, configure x routes
 # -configure VLAN
 # -configure a firewall
 
-# defaults
-topology="pp" # two physical devices on one switch
-	      # topology is desctibed by a list of 1 or more switches separated by commas
-	      # the supported interfaces on a switch are:
-	      # p: a physical port
-	      # v: a virtio-net "backend" port, like dpdkvhostuser or vhost-net (dependig on dataplane)
-	      # V: a virtio-net "frontend" port, like virtio-pci in a VM (not yet implemented)
-	      # P: a patch-port for OVS (not yet implemented)
-queues=1 # queues: Number of queue-pairs (rx/tx) to use per device
-switch="ovs" # switch: Currently supported is: testpmd, ovs, linuxbridge, linuxrouter, vpp
-switch_mode="default" # switch_mode: Currently supported list depends on $switch
-numa_mode="strict" # numa_mode: (for DPDK vswitches only)
-			#strict:    All PMD threads for all phys and virt devices use memory and cpu only 
-			#           from the numa node where the physical adapters are located.           
-			#           This implies the VMs must be on the same node.                        
-			#preferred: Just like 'strict', but the vswitch also has memory and in some cases 
-			#           uses cpu from the non-local NUMA nodes.                               
-			#cross:     The PMD threads for all phys devices use memory and cpu from          
-			#           the local NUMA node, but VMs are present on another NUMA node,        
-			#           and so the PMD threads for those virt devices are also on             
-			#           another NUMA node.                                                    
-overlay="none" # overlay: Currently supported is: none (for all switch types) and vxlan (for linuxbridge and ovs)
-ovs_build="rpm" # either "rpm" or "src"
-dpdk_nic_kmod="vfio-pci" # dpdk-devbind: the kernel module to use when assigning a network device to a userspace program (DPDK application)
+
+
+###############################################################################
+# Default values
+###############################################################################
+
+#
+# topology:
+#
+#	Two physical devices on one switch.  'topology' is desctibed by a list 
+#	of 1 or more switches separated by commas.  The supported interfaces on 
+#	a switch are:
+#
+#	p:		A physical port
+#
+#	v:		A virtio-net "backend" port, like dpdkvhostuser or vhost-net (dependig on dataplane)
```

Comment:
I am a little unsure of what these should stand for.  We have to keep in mind how this is used in a host and a VM (or container eventually).  For example, to bridge two virtio devices in a VM, one you actually need to use a "pp" topology, because from the VM's point of view, these are PCI devices.  For that reason, I don't think "V" should exist here -there is no reason for that distinction in my mind.  The details of the PCI device, like wether it is a xxv710 or a virtio-net, can be determined with further introspection.

Reply:
@k-rister @bmichalo

Reply:
I just pushed an update in my github hopefully clarifying things.

Reply:
For now I think we all agree that "p" means PCI device, any PCI device, even virtio-net, and "v" means a virtio-net backend device, which could be different things in different dataplanes (vhostuser, vhost_net, etc).